### PR TITLE
init: remove flox-extras

### PR DESCRIPTION
**Pull request by tomberek**
Migrate everything from flox-extras into floxpkgs.

### CODE REVIEW

File: flox.nix [hunk 1]
```diff
@@ -24,20 +24,19 @@
   config = {
     extraPlugins = [
       (inputs.capacitor.plugins.allLocalResources {})
-      (inputs.flox-extras.plugins.catalog {})
+      (import ./plugins/catalog.nix { inherit self lib; } {})
```
Comment (mkenigs) [inline flox.nix:27]: (probably not an issue for this PR) but it seems like this is not how we'd want users to use their own plugins in floxpkgs repos

**Review by mkenigs — APPROVED** [review]
Tested that the `floxEnvs` plugin works with this

File: flox.nix (hunk 1 above)
Comment (tomberek) [reply flox.nix:27]: The issue is one of needing to have the plugins in order to evaluate the plugins (bootstrapping and avoiding an infinite fixed point). I suspect it is possible, but not with the current capacitor. So here we are directly bringing in the plugin manually. 

At the moment the recommended syntax has a limitation that “you cannot define a plugin in the same project that uses it”. You are correct that this should be fixed in a separate repo and then this wart removed.

**Comment by tomberek** [comment]
> Tested that the `floxEnvs` plugin works with this

My other main consideration is the testing framework.

**Pull request merged**